adapters/restclient/lib/opentelemetry/adapters/restclient/patches/request.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module RestClient
+      module Patches
+        # Module to prepend to RestClient::Request for instrumentation
+        module Request
+          def execute(&block)
+            return super(&block) if disable_span_reporting?
+
+            trace_request do |_span|
+              super(&block)
+            end
+          end
+
+          # Override implementation (subclass) to determine per-connection
+          # span reporting rules.
+          def disable_span_reporting?
+            false
+          end
+
+          private
+
+          def trace_request # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+            span = tracer.start_span(
+              url,
```

Comment:
This will lead to high-cardinality span names. See: https://github.com/open-telemetry/opentelemetry-ruby/pull/175#discussion_r373245633.

Reply:
@mwear We could use [`"HTTP #{METHOD_NAME}"`](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#name). If that sounds good I'll update both PRs.

Only other thing I can think of that would add specificity without introducing a lot more would be to include URL host/domain.